include/alpaka/dev/DevCpu.hpp

```diff
@@ -99,19 +105,36 @@ namespace alpaka
         template<>
         struct GetName<DevCpu>
         {
-            ALPAKA_FN_HOST static auto getName(DevCpu const& /* dev */) -> std::string
+            ALPAKA_FN_HOST static auto getName(DevCpu const& dev) -> std::string
             {
-                return cpu::detail::getCpuName();
+                auto& devProperties = dev.m_spDevCpuImpl->deviceProperties();
+                std::call_once(
+                    dev.m_spDevCpuImpl->onceFlag(),
+                    [&]() noexcept
+                    {
+                        devProperties->name = cpu::detail::getCpuName();
+                        devProperties->totalGlobalMem = cpu::detail::getTotalGlobalMemSizeBytes();
+                    });
```

Comment:
here, and for the other backends: I would suggest to wrap the common code use to read all device properties in a function, and call it from multiple places, instead of duplicating the code.